schematics/ng-componize/htmlParser.ts

```diff
@@ -0,0 +1,167 @@
+        newComponentTsPath: string,
+        className: string
+      ): string {
+        const project = new Project({});
+
+        const newComponentTsFile = project.createSourceFile(
+          newComponentTsPath,
+          newComponentContent
+        );
+
+        const classNameInTsComponent = ComputeClassName(className);
+
+        const myClass = newComponentTsFile.getClassOrThrow(
+          classNameInTsComponent
+        );
+
+        addInputInImport(newComponentTsFile);
+
+        addInputProperties(parameters, myClass);
+
+        newComponentTsFile.formatText();
+        return newComponentTsFile.getFullText();
+      };
+    },
+  },
+];
+
+export function launchParsing(htmlContent: string): FoundItem[] {
+  let result: FoundItem[] = [];
+
+  for (const configuration of regexpConfigurations) {
+    const foundItems = htmlContent.match(configuration.lookupRegexp);
+
+    if (foundItems && foundItems.length > 0) {
+      const parserFunctionsConfiguration = parserFunctionsConfigurations.filter(
+        (c) => c.names.some((n) => n === configuration.name)
+      )[0];
+      result.push({
+        name: configuration.name,
+        foundItems: foundItems,
+        writeHtmlContent: parserFunctionsConfiguration.htmlUpdater(foundItems),
+        writeTSContent: parserFunctionsConfiguration.computeNewComponentTs(
+          foundItems
+        ),
+      });
+    }
+  }
+
+  return result;
+}
+
+export function extractParameterName(param: string): string {
+  return param
+    .slice(
+      param.lastIndexOf('.') + 1,
+      param.indexOf('|') >= 0 ? param.indexOf('|') : param.length
+    )
+    .replace('$', '')
+    .trim();
+}
+
+interface RegexpConfiguration {
+  name: ConfigurationName;
+  description: string;
+  lookupRegexp: RegExp;
+}
+
+interface ParserFunctionsConfiguration {
+  names: ConfigurationName[];
+  htmlUpdater(parameter: string[]): () => string;
+  computeNewComponentTs(
+    paramater: string[]
+  ): (newComponentContent: string, path: string, className: string) => string;
+}
+
+const regexpConfigurations: RegexpConfiguration[] = [
+  {
+    name: ConfigurationName.InterpolationLookup,
+    description: 'Looking for Interpolated string',
+    lookupRegexp: /(?<={{\s*)([a-zA-Z\-0-9]*)(?=\s*}})/g,
+  },
+  {
+    name: ConfigurationName.BindingLookup,
+    description: 'Looking for data-bindind',
+    lookupRegexp: /(?<=\[[0-9a-zA-Z\-]*\]=\")([\s*\|\?\.\$0-9a-zA-Z\-]*)(?=\")/g,
+  },
+  {
+    name: ConfigurationName.AngularDireciveLookup,
+    description: 'Looking for angular directives',
+    lookupRegexp: /(?<=\*ng[a-zA-Z]*=\")([\s*\|\?\.\$0-9a-zA-Z\-]*)(?=\")/g,
+  },
+];
+
+function addInputProperties(
+  parameters: string[],
+  myClass: ClassDeclaration
+): void {
+  parameters.forEach((parameter) => {
+    myClass.insertProperty(0, {
+      decorators: [
+        {
+          name: 'Input()',
+        },
+      ],
+      name: extractParameterName(parameter),
+      type: 'any',
```

Comment:
this is worth a future PR 👍

Reply:
can you make your comment clearer ? no sure to understand